### ISSUE TITLE
Identify tenant

### ISSUE THREAD

**Issue by IvoVermeer** [dbeck121/ConVista-CPI-Helper-Chrome-Extension#7]
Hi Doiminic,

I was working on something to update the tabname when I came accross your extenstion. As that feature is on your todolist, I thought it would be better to add it into your extension. I have a new branch ready in my fork. The branch name is doc-title.

Please take a look and maybe give it a test.

**Comment by IvoVermeer** [comment]
I forgot to include a link to the branch, here it is:
https://github.com/IvoVermeer/ConVista-CPI-Helper-Chrome-Extension/tree/doc-title

**Comment by dbeck121** [comment]
Hi,

thank you for contributing to the CPI Helper. Indeed, this was on the todo-list. So let's make version 1.2 out of it. 

I tried to run it but the form in the popup sais "undefined" and when I canged it, it went back to undefined after a second.

![grafik](https://user-images.githubusercontent.com/54773577/79214782-44321000-7e4b-11ea-9241-d9355748e8da.png)

Also Last Visited IFlows do not work anymore.

Maybe we can structure the PopUp like

<h3>Tenant <original tenant name>
Set custom name for tab <your input here>
<h3>Last visited on this tenant
....
<h3>General Links

Can you fix the bugs and send a pull request to the same branch that I created?
I will adjust the version and readme and we can release a new Version than.

Best regards
Dominic

**Comment by IvoVermeer** [comment]
Hi Dominic,

I think we can later merge my doc-title into any branch on your repo, so I'll keep working on my branch for now. I'll make sure to update my repo with changes in yours so nothing's missed.

I don't have the issue you're having. Do you maybe see any errors in the CPI page's console, the extension's background-page or in the popup window's console?

I did have the issue earlier this morning buth that was due to the message handler not being setup in the contentScript.js and that's now solved.

**Comment by dbeck121** [comment]
thats strange because the error disappeared. But I did not pull new data. 

I will try it again later but now it works pretty good.

I was thinking about changing the favicon too because my browser offen looks like:

![grafik](https://user-images.githubusercontent.com/54773577/79240598-48bdef00-7e72-11ea-8086-e63b2dc71ac2.png)

I have this snippet here to change an icon:

```
  var link = document.querySelector("link[rel*='icon']") || document.createElement('link');
    link.type = 'image/x-icon';
    link.rel = 'shortcut icon';
    link.href = chrome.extension.getURL("/images/favicons/red.ico");
    document.getElementsByTagName('head')[0].appendChild(link);

and in manifest
  "web_accessible_resources": [
    "images/favicons/*.ico"
	],
```
So if we have maybe 4 or 5 icons in the favicons folder, we could add a dropdown to select one of these. What do you think?

**Comment by dbeck121** [comment]
I played a bit with the idea:
![grafik](https://user-images.githubusercontent.com/54773577/79242875-25e10a00-7e75-11ea-9592-16526896d2b3.png)

![grafik](https://user-images.githubusercontent.com/54773577/79242952-4741f600-7e75-11ea-9ee5-fd6f92791cf1.png)

**Comment by IvoVermeer** [comment]
Hi Dominic,

Yes, that's a great idea. I'm going to add something for that. I was also thinking of injecting some extra CSS and change the background color of the header. That's either .sapTntToolHeader or #shell--toolHeader. I'll play around with that and let you know when there's something to show.

I'm doing this in my spare time, so I have limited time to spend on it.

**Comment by dbeck121** [comment]
Great idea with the header.
You have all the time you need. So does not matter if we release it next week or in a month.
I uploaded a few favicons for you in /images/favicons if you want to use them but don't feel pushed. If you have a better idea.

**Comment by IvoVermeer** [comment]
I can't get the icon to change. Can you share your code snippet that you used to make it work? I see the new <link> added to the document, but it's not showing the icon in the tab.

I do now have the title and header color working. I re-wrote the entire thing en put the content-side of the scripting in a seperate file. I'm running it through some tests today and I'll push it to my fork tonight

**Comment by dbeck121** [comment]
`function setTitle(title) {
  let intervalId;
  document.title = title;
  let count = 5
  // set title again aftet 2sec
  intervalId = setInterval((title) => {
      count--;
      document.title = title;
	 
    **var link = document.querySelector("link[rel*='icon']") || document.createElement('link');
    link.type = 'image/x-icon';
    link.rel = 'shortcut icon';
    link.href = chrome.extension.getURL("/images/favicons/2.png");
    document.getElementsByTagName('head')[0].appendChild(link);**
	
      if (count == 0) {
          clearInterval(intervalId);
      }
  }, 2000, title)
}
`

This is your code with the addition to change the favicon. I also had to update the manifest.json file

`{
  "name": "ConVista SAP CPI Helper",
  "short_name": "CPI Helper",
  "version": "1.1.1",
  "description": "Extends the SAP Cloud Platform Integration with some useful features to improve usability.",
  "author": "Dominic Beckbauer @ Convista Consulting AG",
  "homepage_url": "https://github.com/dbeck121/ConVista-CPI-Helper-Chrome-Extension",
  "permissions": [
    "declarativeContent",
    "storage",
    "https://*.hana.ondemand.com/itspaces",
    "webRequest"
  ],
  "background": {
    "scripts": [
      "background.js",
      "hot-reload.js"
    ],
    "persistent": true
  },
  "content_scripts": [
    {
      "matches": [
        "https://*.hana.ondemand.com/itspaces/*"
      ],
      "js": [
        "/lib/prettify/prettify.js",
        "/lib/jszip/jszip.min.js",
        "contentScript.js"
      ],
      "css": [
        "/lib/prettify/prettify.css",
        "contentScript.css"
      ]
    }
  ],
  "page_action": {
    "default_popup": "popup.html",
    "default_icon": {
      "16": "images/cpi_helper16.png",
      "32": "images/cpi_helper32.png",
      "48": "images/cpi_helper48.png",
      "128": "images/cpi_helper128.png"
    }
  },
  "icons": {
    "16": "images/cpi_helper16.png",
    "32": "images/cpi_helper32.png",
    "48": "images/cpi_helper48.png",
    "128": "images/cpi_helper128.png"
  },
  **"web_accessible_resources": [
    "images/favicons/*.png"
	],**
	
  "manifest_version": 2
}`

I hope that helps.

**Comment by IvoVermeer** [comment]
Everything works now. I put everything into a new branch to prevent conflicts with your doc-title branch.

Sometime the popup seems to have lost the connection to the content scripts. I don't know who to fix that. I also made sure that if you have multiple tabs open for the same tenant(even accross multiple windows) that all tabs get updated when changing the settings. Check the monitorSyncStore function for that.

Can you add this to your browser and see if it works ok on your end too?

Here is my branch: https://github.com/IvoVermeer/ConVista-CPI-Helper-Chrome-Extension/tree/identify-tenant

Once your satisfied, I'll creat a PR. Don't create the same branch though:)

**Comment by dbeck121** [comment]
that looks pretty good. I like your idea with the color scheme :-)

Maybe one last thing. Default favicon does not work anymore if you change it back. Maybe lets save the original favicon from sap also in the favicon folder. I think we need this because we have overwritten it.

Just create a PR when you are done. It has become a pretty nice feature. I will create version 1.2.0 than. Maybe I will write a small blog post. If you give me your username in the sap community, I will link your name. the fame is yours ;)

**Comment by bdbais** [comment]
Could you add Global Account  information on CPI title?  

SAP Cloud Platform Integration ; Global Account: : Customer  - SubAccount : PROD

**Comment by dbeck121** [comment]
What do you mean with title? The label of the tab?
Unfortunately I can't access information of the underlying scp but you can write it in the tab label. Does that help you?

**Comment by IvoVermeer** [comment]
> that looks pretty good. I like your idea with the color scheme :-)
> 
> Maybe one last thing. Default favicon does not work anymore if you change it back. Maybe lets save the original favicon from sap also in the favicon folder. I think we need this because we have overwritten it.
> 
> Just create a PR when you are done. It has become a pretty nice feature. I will create version 1.2.0 than. Maybe I will write a small blog post. If you give me your username in the sap community, I will link your name. the fame is yours ;)

Hi Dominic,

I think I have a fix, but I had to add the default icon to the repository. So if SAP ever updates their icon, you will have to update your repo too.

I'll test it some more today and then create a PR.

**Comment by dbeck121** [comment]
Hi @IvoVermeer ,
any news?

**Comment by dbeck121** [comment]
sry, wrong button. I did not want to close this

**Comment by IvoVermeer** [comment]
Hi @dbeck121 ,

The new is not good; From time-to-time I see an error in the extension-page. The MutationObserver sometime seems to lose its connection to the document. I'm trying to find the proper solution.

Are you also testing my commits? Are you running into the same error?

**Comment by dbeck121** [comment]
Sorry for the late response, I will have a look the next days.

**Comment by IvoVermeer** [comment]
Hi Dominic,

I think I have a solution. I'm working in CPI all day today, so I can test it too. I'll push a new commit to my fork by the end of the day so you can then test it too.

**Comment by IvoVermeer** [comment]
Hi Dominic,

I've created the PR, I think this issues should now be closed?

**Comment by dbeck121** [comment]
Great! Sorry for my late response the last days. There is much to do. I will test it and release it these days :-)
I will also write a blog post and mention you as contributor if that is ok.

**Comment by IvoVermeer** [comment]
No problem, let me know if you need any input for that blog

Also, please review the code I wrote and let me know if you have my questions.

**Comment by dbeck121** [comment]
I just released 1.2.0 with your code changes. Thank you again for the contribution. I think your code looks a bit more professional than mine :-)

I will write a blog post in the sap community on monday :-)

**Comment by dbeck121** [comment]
https://blogs.sap.com/2020/05/11/cpi-working-on-multiple-cpi-tenants-do-not-lose-the-overview-with-convista-cpi-helper

:-)
